### PR TITLE
XW-4232 | rendering featured video outside render - AB test

### PR DESCRIPTION
**Pull request by latata**
## Links

* https://wikia-inc.atlassian.net/browse/XW-4232

## Description

All if statements with `true`/`false` as condition will be replaced with AB test variation check - it's just for testing now

## Reviewers

@Wikia/x-wing 

DON'T MERGE
Revert c63da906752944fb2ff3fc48100bbca412e30780 commit before merging

### CODE REVIEW

File: app/components/application-wrapper.js [hunk 1]
```diff
@@ -64,6 +67,14 @@ export default Component.extend({
 		return `${vertical}-vertical`;
 	}),
 
+	fandomAppSmartBannerObserver: observer('isFandomAppSmartBannerVisible', () => {
+		if (true || inGroup('FEATURED_VIDEO_VIEWABILITY_VARIANTS', 'RENDER_ORDER')) {
```
Comment (hakubo) [inline app/components/application-wrapper.js:71]: remember to remove 'true'

File: app/components/application-wrapper.js [hunk 2]
```diff
@@ -64,6 +67,14 @@ export default Component.extend({
...
+			scheduleOnce('afterRender', () => {
```
Comment (hakubo) [inline app/components/application-wrapper.js:72]: recently we've learned that this afterRender might not happen when you think it'll happen
but for a test it should be OK

File: app/components/article-featured-video.js [hunk 3]
```diff
@@ -31,9 +37,36 @@ export default Component.extend({
 	// when navigating from one article to another with video, we need to destroy player and
 	// reinitialize it as component itself is not destroyed. Could be done with didUpdateAttrs
 	// hook, however it is fired twice with new attributes.
-	videoIdObserver: on('didInsertElement', observer('model.embed.jsParams.videoId', function () {
-		this.destroyVideoPlayer();
-		this.initVideoPlayer();
+	videoIdObserver: on('didInsertElement', observer('articleId', function () {
+		if (false && !inGroup('FEATURED_VIDEO_VIEWABILITY_VARIANTS', 'RENDER_ORDER')) {
```
Comment (hakubo) [inline app/components/article-featured-video.js:41]: remember to remove false

File: app/modules/abtest/featured-video-render-order-helper.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+		M.loadScript('/mobile-wiki/assets/mobile-wiki.js', false, false, 'anonymous');
+	}
+}
+
+export function setFeaturedVideoPlayer(player) {
+	window.featuredVideoPlayer = player;
+}
+
+function createCookie(name, value, days = 14) {
+	let expires = "";
```
Comment (hakubo) [inline app/modules/abtest/featured-video-render-order-helper.js]: ''

File: app/modules/dom-helper.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class DomHelper {
```
Comment (hakubo) [inline app/modules/dom-helper.js]: why class? 
could this be a simple module with one export?

**Review by hakubo — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

File: app/index.html [hunk 6]
```diff
@@ -44,6 +45,12 @@
 	}
 </script>
 
+<script>
+	if (true) {
```
Comment (bkoval) [inline app/index.html:49]: Do we need the if at all?

File: app/modules/abtest/featured-video-render-order-helper.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+	player.on('captionsSelected', ({selectedLang}) => {
+		createCookie('featuredVideoCaptions', selectedLang);
+	});
+
+	player.once('adImpression', () => {
+		initializeMobileWiki();
+	});
+
+	player.once('videoStart', () => {
```
Comment (bkoval) [inline app/modules/abtest/featured-video-render-order-helper.js:51]: Should we also initialize mobile wiki on "setupError"?

File: app/modules/abtest/featured-video-render-order-helper.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+		initializeMobileWiki();
+	});
+
+	// initialize mobile wiki after 8 seconds if it hasn't been initialized yet
+	setTimeout(() => {
```
Comment (bkoval) [inline app/modules/abtest/featured-video-render-order-helper.js:60]: If we initialize mobile-wiki on setupError of JW we should have everything covered, right? Then we could drop this fallback

File: app/modules/abtest/featured-video-render-order-helper.js [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+		.then((new JWPlayerVideoAds(params)).getConfig())
+		.then(initializePlayer.bind(this, params));
+}
+
+export function loadJWPlayerAssets(params) {
+	const head = document.getElementsByTagName('head')[0],
+		link = document.createElement('link');
+	link.rel = 'stylesheet';
+	link.type = 'text/css';
+	link.href = '/mobile-wiki/assets/jwplayer/index.css';
```
Comment (bkoval) [inline app/modules/abtest/featured-video-render-order-helper.js]: Could we just add a style tag to index.html?

File: app/modules/abtest/featured-video-render-order-helper.js [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+		updateFeaturedVideoPosition();
+	}, 'anonymous');
+}
+
+export function getQueryParameterByName(name) {
+	name = name.replace(/[\[\]]/g, "\\$&");
+	const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+		results = regex.exec(window.location.href);
+	if (!results) return null;
+	if (!results[2]) return '';
```
Comment (bkoval) [inline app/modules/abtest/featured-video-render-order-helper.js]: Wow, this is magic.

**Review by bkoval — APPROVED** [review]
Great job!

File: app/modules/abtest/featured-video-render-order-helper.js (hunk 8 above)
Comment (latata) [reply app/modules/abtest/featured-video-render-order-helper.js:60]: I'd leave it for unknown scenarios ;)

**Comment by qaga** [comment]
Code from that PR was already merged to feature branch.

**Pull request closed without merging**